model_archiver/model_packaging_utils.py

```diff
@@ -0,0 +1,288 @@
+            _params.add(tensor_vals.name)
+
+        input_data = []
+        for graph_input in graph.input:
+            shape = []
+            if graph_input.name not in _params:
+                for val in graph_input.type.tensor_type.shape.dim:
+                    shape.append(val.dim_value)
+                input_data.append((graph_input.name, tuple(shape)))
+
+        sym, arg_params, aux_params = onnx_mxnet.import_model(os.path.join(model_path, onnx_file))
+        # UNION of argument and auxillary parameters
+        params = dict(arg_params, **aux_params)
+        # rewrite input data_name correctly
+        with open(os.path.join(model_path, signature_file), 'r') as f:
+            data = json.loads(f.read())
+            data['inputs'][0]['data_name'] = input_data[0][0]
+        with open(os.path.join(model_path, signature_file), 'w') as f:
+            f.write(json.dumps(data, indent=2))
+
+        with open(os.path.join(model_path, symbol_file), 'w') as f:
+            f.write(sym.tojson())
+
+        save_dict = {('arg:%s' % k): v.as_in_context(mx.cpu()) for k, v in params.items()}
+        mx.nd.save(os.path.join(model_path, params_file), save_dict)
+        return symbol_file, params_file
+
+    @staticmethod
+    def generate_publisher(publisherargs):
+        publisher = Publisher(author=publisherargs.author, email=publisherargs.email)
+        return publisher
+
+    @staticmethod
+    def generate_engine(engineargs):
+        engine = Engine(engine_name=engineargs.engine)
+        return engine
+
+    @staticmethod
+    def generate_model(modelargs):
+        model = Model(model_name=modelargs.model_name, handler=modelargs.handler)
+        return model
+
+    @staticmethod
+    def generate_manifest_json(args):
+        """
+        Function to generate manifest as a json string from the inputs provided by the user in the command line
+        :param args:
+        :return:
+        """
+        arg_dict = vars(args)
+
+        publisher = ModelExportUtils.generate_publisher(args) if 'author' in arg_dict and 'email' in arg_dict else None
+
+        engine = ModelExportUtils.generate_engine(args) if 'engine' in arg_dict else None
+
+        model = ModelExportUtils.generate_model(args)
+
+        manifest = Manifest(runtime=args.runtime, model=model, engine=engine, publisher=publisher)
+
+        return str(manifest)
+
+    @staticmethod
+    def clean_temp_files(temp_files):
+        for f in temp_files:
+            os.remove(f)
+
+    @staticmethod
+    def zip(export_file, model_name, model_path, files_to_exclude, manifest):
+        mar_path = os.path.join(export_file, '{}{}'.format(model_name, MODEL_ARCHIVE_EXTENSION))
+        with zipfile.ZipFile(mar_path, 'w', zipfile.ZIP_DEFLATED) as z:
+            ModelExportUtils.zip_dir(model_path, z, set(files_to_exclude))
+            # Write the manifest here now as a json
+            z.writestr(os.path.join(MAR_INF, MANIFEST_FILE_NAME), json.dumps(manifest, indent=4))
```

Comment:
Is this using utf-8 encoding

Reply:
Done. a6046e1a4a73100b4da6abe9c0745575853fdc7a